pkg/types/powervs/validation/machinepool.go

```diff
@@ -0,0 +1,67 @@
+package validation
+
+import (
+	"math"
+	"regexp"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate VolumeIDs
+	for i, volumeID := range p.VolumeIDs {
+		_, err := uuid.Parse(volumeID)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeIDs").Index(i), volumeID, "volume ID must be a valid UUID"))
+		}
+	}
+
+	// Validate Memory
+	if p.Memory != "" {
+		memory, err := strconv.ParseInt(p.Memory, 10, 64)
+		if err == nil {
+			if memory < 2 || memory > 64 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be from 2 to 64 GB"))
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "memory must be an integer number of GB that is at least 2 and no more than 64"))
+		}
+	}
+
+	// Validqte Processors
+	if p.Processors != "" {
+		processors, err := strconv.ParseFloat(p.Processors, 64)
+		if err == nil {
+			if processors < 0.25 || processors > 32 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("processors"), p.Processors, "number of processors must be from .25 to 32 cores"))
+			}
+			if math.Mod(processors*1000, 2) == 0 && math.Mod(processors*100, 25) == 0 {
```

Comment:
.5 is valid. maybe that statement needs some tweaking to be more clear? since those both result in zeros and the lhs and rhs are both true: true && true == true 
to be sure, i did make a little test program:
```
func main() {
        processors := []float64{0.0, 0.5, 0.25, 0.6, 1.4, 1.5, 2.0, 4, 5, 32, 16.111, 100.999}
        for _, n := range processors {
                valid := math.Mod(n*1000, 2) == 0 && math.Mod(n*100, 25) == 0
                fmt.Println(fmt.Sprintf("%f: %v", n, valid))
        }
}

Reply:
But then it shouldn't report an error if it's valid?

Reply:
ah! now i see your point. will fix. thanks